js/shopping-list.js

```diff
@@ -0,0 +1,50 @@
+var ulElement = document.querySelector('ul');
+
+function addListItems(event){
+    var form = event.target;
+    var items = form.querySelector('#items').value.split(',');
+
+    if(items[0].length > 0){
+        document.querySelector('#user-warning').textContent = "";
+        for(var i = 0; (i < items.length) && (items[i].length > 0) && (items[i] !== ' '); i++){
+            appendItemToList(createXImgElement(), items[i]);
+        }
+    } else {
+        document.querySelector('#user-warning').textContent = "Please enter an item.";
+    }
+    event.preventDefault();
+    form.reset();
+}
+
+function createXImgElement(){
+    var imgElement = document.createElement('img');
+    imgElement.setAttribute('src', 'media/x-icon.png');
+    imgElement.setAttribute('alt', 'A red x to indicate removal.');
+    imgElement.addEventListener('click', removeListItem);
+    return imgElement;
+}
+
+function appendItemToList(imgElement, item){
```

Comment:
We could simplify this method a little by removing the `imgElement` parameter and instead changing line 30 to `liElement.appendChild(createXImgElement());`. Since we're always appending the same return value, there's no need for it to be a parameter - and since we only reference it on that one line, there's no need to cache it.